library/azure_rm_resource.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_resource
+version_added: "2.6"
+short_description: Create any Azure resource.
+description:
+  - Create any Azure resource using Azure REST API.
+
+options:
+  url:
+    description:
+      - Azure RM Resource URL.
```

Comment:
add more properties like required?

Reply:
pls add properties like required or not. choices if there's and type.

Reply:
if not required it should be omitted, only if required should be "required: yes".

Reply:
also same applies to type, if type is string, then it's default and no need to include in the documentation.